packages/site/src/content/InputFile/InputFileNotes.mdx

```diff
@@ -118,7 +126,194 @@ allowed to be uploaded.
+
+### Dropzone variation (default)
+
+```tsx
+// size="base" (default)
+<InputFile getUploadParams={getUploadParams}>
+  <InputFile.DropzoneWrapper>
+    <Content spacing="small">
+      <InputFile.Button size="small" fullWidth={false} />
+      <InputFile.HintText />
+      <InputFile.Description />
+    </Content>
+  </InputFile.DropzoneWrapper>
+</InputFile>
+
+// size="small"
+<InputFile size="small" getUploadParams={getUploadParams}>
+  <InputFile.DropzoneWrapper>
+    <Content spacing="small">
+      <InputFile.Button size="small" fullWidth={false} />
+      {/* HintText and Description are omitted in small size */}
+    </Content>
+  </InputFile.DropzoneWrapper>
+</InputFile>
+```
+
+### Button variation
+
+```tsx
+<InputFile variation="button" getUploadParams={getUploadParams}>
+  <InputFile.Button fullWidth={true} />
+</InputFile>
+```
+
+## Customization
+
+### Using props
+
+Basic customization can be achieved through props:
+
+```tsx
+<InputFile
+  getUploadParams={getUploadParams}
+  size="small"
+  buttonLabel="Choose Images"
+  allowMultiple={true}
+  allowedTypes="images"
+  hintText="Drop your images here"
+  description="Supported formats: JPG, PNG"
+/>
+```
+
+### Using children
+
+For complete control over the layout, you can provide custom children:
+
+```tsx
+<InputFile getUploadParams={getUploadParams}>
+  <InputFile.DropzoneWrapper>
+    <h2>Upload Your Files</h2>
+    <InputFile.Button size="large" />
+    <InputFile.HintText />
+    <InputFile.Description />
+  </InputFile.DropzoneWrapper>
+</InputFile>
+```
+
+### Available subcomponents
+
+1. `InputFile.Button`: Upload button
+
+   ```tsx
+   <InputFile.Button size="large" fullWidth={false} variation="work" />
+   ```
+
+2. `InputFile.HintText`: Helper text for the dropzone
+
+   ```tsx
+   <InputFile.HintText size="small" />
+   ```
+
+3. `InputFile.Description`: Additional description text
+
+   ```tsx
+   <InputFile.Description size="small" variation="subdued" />
+   ```
+
+4. `InputFile.DropzoneWrapper`: Wrapper for dropzone content
+   ```tsx
+   <InputFile.DropzoneWrapper>{/* Your content */}</InputFile.DropzoneWrapper>
+   ```
+
+### Using context
+
+Subcomponents automatically access the InputFile context. The context includes:
+
+```tsx
+interface InputFileContentContextValue {
+  fileType: string; // Type of file being uploaded
+  allowMultiple: boolean; // Multiple files allowed
+  description?: string; // Description text
+  hintText?: string; // Hint text
+  buttonLabel?: string; // Button label
+  size: "small" | "base"; // Component size
+}
+```
+
+You can also access this context in custom components:
+
+```tsx
+import { useInputFileContentContext } from "./InputFileContentContext";
+
+function CustomUploadContent() {
+  const { fileType, allowMultiple, size } = useInputFileContentContext();
+
+  return (
+    <div>
+      <h3>Upload {allowMultiple ? `${fileType}s` : fileType}</h3>
+      <InputFile.Button size={size} />
+    </div>
+  );
+}
```

Comment:
I think showing an explicit usage of this inside an `InputFile` would make sense. I know we have the example in storybook but people may not check both places